app/lib/features/spaces/widgets/public_spaces_selector.dart

```diff
@@ -336,6 +336,86 @@ class PublicSpaceItem extends ConsumerWidget {
   }
 }
 
+class QuickActionWidget extends ConsumerWidget {
+  final OnSelectedFn onSelected;
+  final OnSelectedMatchFn? onSelectedMatch;
+  final bool canMatchAlias;
+  final bool canMatchId;
+  const QuickActionWidget({
+    super.key,
+    required this.onSelected,
+    this.onSelectedMatch,
+    this.canMatchAlias = false,
+    this.canMatchId = false,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final searchVal = ref.watch(searchValueProvider);
+    if (onSelectedMatch != null && searchVal?.isNotEmpty == true) {
+      final aliased = RegExp(r'https://matrix.to/#/(?<alias>#.+):(?<server>.+)')
```

Comment:
As I am reviewing it commit by commit for better analysis, these patterns should exist common-wide as utilities but if its been moved in later commits, that's good. 👍🏻